asv_bench/benchmarks/rolling.py

```diff
@@ -6,43 +6,48 @@
 import pandas as pd
 import xarray as xr
 
-from . import randn
-
+from . import randn, requires_dask
 
 nx = 3000
 ny = 2000
 nt = 1000
 window = 20
-ds = xr.Dataset({'var1': (('x', 'y'), randn((nx, ny), frac_nan=0.1)),
-                 'var2': (('x', 't'), randn((nx, nt))),
-                 'var3': (('t', ), randn(nt))},
-                coords={'x': np.arange(nx),
-                        'y': np.linspace(0, 1, ny),
-                        't': pd.date_range('1970-01-01', periods=nt, freq='D'),
-                        'x_coords': ('x', np.linspace(1.1, 2.1, nx))})
-
-
-def time_rolling(func, center):
-    getattr(ds.rolling(x=window, center=center), func)()
 
 
-time_rolling.param_names = ['func', 'center']
-time_rolling.params = (['mean', 'count'], [True, False])
+class Rolling(object):
+    def setup(self, *args, **kwargs):
+        self.ds = xr.Dataset(
+            {'var1': (('x', 'y'), randn((nx, ny), frac_nan=0.1)),
+             'var2': (('x', 't'), randn((nx, nt))),
+             'var3': (('t', ), randn(nt))},
+            coords={'x': np.arange(nx),
+                    'y': np.linspace(0, 1, ny),
+                    't': pd.date_range('1970-01-01', periods=nt, freq='D'),
+                    'x_coords': ('x', np.linspace(1.1, 2.1, nx))})
 
+    def time_rolling(self, func, center):
+        getattr(self.ds.rolling(x=window, center=center), func)()
 
-def time_rolling_np(window_, min_periods):
-    ds.rolling(x=window_, center=False, min_periods=min_periods).reduce(
-        getattr(np, 'nanmean'))
+    time_rolling.param_names = ['func', 'center']
+    time_rolling.params = (['mean', 'count'], [True, False])
```

Comment:
Maybe we should write our own `@parameterized` decorator, as suggested in https://github.com/airspeed-velocity/asv/issues/482?